apps/language_server/lib/language_server/server.ex

```diff
@@ -320,6 +322,14 @@ defmodule ElixirLS.LanguageServer.Server do
     |> Enum.uniq()
     |> WorkspaceSymbols.notify_uris_modified()
 
+    formatter_changed? = Enum.any?(changes, fn %{"uri" => uri} ->
+      Path.basename(uri) == ".formatter.exs"
```

Comment:
Is it more complicated than this? I realize this could lead to some false positives but that isn't really a big issue IMHO. Are there files beyond this file that can be allowed to affect formatting?

Reply:
How about `.formatter.exs` in subdirectories? AFAIK it is supported in subapps in umbrella and there's an unfinished PR that tries to make it work in more cases https://github.com/elixir-lsp/elixir-ls/pull/362
Though editing `.formatter.exs` is likely rare I can see that happen more often when switching between branches

Reply:
@lukaszsamson that will work just fine here. What this code is doing is saying "If any files are changed anywhere in the open project named ".formatter.exs" then rebuild the cache.

I do suppose the following would be an issue: If you `cd` into an elixir project's `lib` folder and `code .` there, it won't be watching the `.formatter.exs` in the upper directory. Let me see if that is honored today though, if not then this at least isn't a regression.

Reply:
OK, I didn't notice ``Path.basename` call.

Regarding the `cd` to `lib` - we don't support non mix projects.

Reply:
`cd` into `lib` might be unsupported, but what about `cd` into `apps/myapp` within an umbrella? A single app within an umbrella is still considered a mix project afaik.

Reply:
Awesome, that definitely simplifies things. There are still a few bugs I'm working out, hope to have this in a ready state by early next week.